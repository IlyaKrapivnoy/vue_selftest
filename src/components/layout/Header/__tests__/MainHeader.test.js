import { mount } from "@vue/test-utils";
import NavMenu from "@/components/layout/Header/MainHeader.vue";

describe("NavMenu.vue", () => {
  it("renders without errors", () => {
    const wrapper = mount(NavMenu);

    expect(wrapper.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = mount(NavMenu);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
