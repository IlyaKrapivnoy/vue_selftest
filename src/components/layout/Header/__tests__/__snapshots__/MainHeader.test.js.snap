// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu.vue matches snapshot 1`] = `
"<main class="container mx-auto">
  <el-menu default-active="1" class="el-menu-demo flex justify-between items-center uppercase h-[80px]" mode="horizontal">
    <router-link to="/">
      <el-menu-item index="1" class="logo-nav-item">
        <div class="flex flex-col items-center cursor-pointer ml-4 mb-2">
          <div class="text-3xl font-bold text-indigo-300"><img src="/logo.png" alt="logo" class="w-[80px]"></div>
          <div class="text-[10px] text-gray-500 -mt-3 leading-4"> practice inc. </div>
        </div>
      </el-menu-item>
    </router-link>
    <div class="flex">
      <router-link to="/">
        <el-menu-item index="0">Home</el-menu-item>
      </router-link>
      <router-link to="todo">
        <el-menu-item index="1">Todo</el-menu-item>
      </router-link>
      <router-link to="quotes">
        <el-menu-item index="2">Quote</el-menu-item>
      </router-link>
      <router-link to="quiz">
        <el-menu-item index="3">Quiz</el-menu-item>
      </router-link>
      <router-link to="band-names">
        <el-menu-item index="4">Random Names</el-menu-item>
      </router-link>
      <router-link to="counter">
        <el-menu-item index="5">Counter</el-menu-item>
      </router-link>
    </div>
  </el-menu>
</main>"
`;
